Plots/MapProjections/NCL_sat_1.py

```diff
@@ -0,0 +1,380 @@
+"""
+NCL_sat_1.py
+===============
+This script illustrates the following concepts:
+   - Creating an orthographic projection
+   - Drawing line contours over a satellite map
+   - Manually labeling contours
+   - Transforming coordinates
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: https://www.ncl.ucar.edu/Applications/Scripts/sat_1.ncl
+    - Original NCL plot: https://www.ncl.ucar.edu/Applications/Images/sat_1_lg.png
+"""
+
+###############################################################################
+# Import packages:
+import xarray as xr
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import numpy as np
+from sklearn.cluster import DBSCAN
+import warnings
+
+import geocat.datafiles as gdf
+import geocat.viz.util as gvutil
+
+###############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine and
+# load the data into xarrays
+ds = xr.open_dataset(gdf.get("netcdf_files/slp.1963.nc"), decode_times=False)
+
+# Get data from the 24th timestep
+pressure = ds.slp[24, :, :]
+
+# Translate short values to float values
+pressure = pressure.astype('float64')
+
+# Convert Pa to hPa data
+pressure = pressure*0.01
+
+# Fix the artifact of not-shown-data around 0 and 360-degree longitudes
+wrap_pressure = gvutil.xr_add_cyclic_longitudes(pressure, "lon")
+
+###############################################################################
+
+
+def findLocalExtrema(da, highVal=1040, lowVal=975, eType='Low'):
+    """
+    Utility function to find local low/high field variable coordinates on a contour map. To classify as a local high, the data
+    point must be greater than highVal, and to classify as a local low, the data point must be less than lowVal.
+    Args:
+        da: (:class:`xarray.DataArray`):
+            Xarray data array containing the lat, lon, and field variable (ex. pressure) data values
+        highVal (:class:`int`):
+            Data value that the local high must be greater than to qualify as a "local high" location.
+            Default highVal is a pressure value of 1040 hectopascals.
+        lowVal (:class:`int`):
+            Data value that the local low must be less than to qualify as a "local low" location.
+            Default lowVal is a pressure value of 975 hectopascals.
+        eType (:class:`str`):
+            'Low' or 'High'
+            Determines which extrema are being found- minimum or maximum, respectively.
+            Default eType is 'Low'.
+    Returns:
+        clusterExtremas (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify local low/high locations
+    """
+
+    # Create a 2D array of coordinates in the same shape as the field variable data
+    # so each coordinate is easily mappable to a data value
```

Comment:
I think what you are doing here is creating a 1D array of X,Y tuples, such that you have a coordinate pair for each grid point. Is that correct? Can you claify?